client/nsxt_client.go

```diff
@@ -88,3 +89,32 @@ func (c *nsxtClient) GetLogicalSwitchState(lswitchID string) (manager.LogicalSwi
 	logicalSwitchesStatus, _, err := c.apiClient.LogicalSwitchingApi.GetLogicalSwitchState(c.apiClient.Context, lswitchID)
 	return logicalSwitchesStatus, err
 }
+
+func (c *nsxtClient) ListAllTransportZones() ([]manager.TransportZone, error) {
```

Comment:
this is actually good, as we abstract `cursor` from the caller and save more lines there. our usecase of localVarOptionals is to pass cursor. when we find other usecases in the future, we can implement one more method that accepts localVarOptionals, and replace line 99 with the new method.

we can refactor other collectors with this as well

Reply:
when we refactor other collectors, I was thinking if we can even make this more DRY, by having common function to iterate the API pages with cursor, the function can accept the API methods that we need, wdyt?

Reply:
Yes that would be great, these cursor loops are everywhere now